cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -412,17 +414,20 @@ public ApplicationWithPrograms deployAppAndArtifact(NamespaceId namespace, @Null
   public ApplicationWithPrograms deployApp(NamespaceId namespace, @Nullable String appName, @Nullable String appVersion,
                                            Id.Artifact artifactId,
                                            @Nullable String configStr,
-                                           ProgramTerminator programTerminator) throws Exception {
-    return deployApp(namespace, appName, appVersion, artifactId, configStr, programTerminator, null);
+                                           ProgramTerminator programTerminator,
+                                           boolean updateSchedules) throws Exception {
+    return deployApp(namespace, appName, appVersion, artifactId, configStr, programTerminator, null, updateSchedules);
```

Comment:
How about keeping this deployApp as it is and passing true from here so that users who don't want to deal with optional spec like ownerPrincipal and schedule can use this one.